src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/Designer/DataTransferTypes.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Debug.Designer
+{
+    internal sealed record PropertyValueChangeRequest(
+        string ProfileName,
+        string PropertyName,
+        object Value);
+
+    internal sealed record Profile(string Name, string DisplayName);
+
+    internal sealed record Category(string Name, string DisplayName);
+
+    internal sealed record PropertyEditor(string Name, ImmutableDictionary<string, string> Metadata);
+
+    internal sealed record PropertyMetadata(
+        string Name,
+        string DisplayName,
+        Profile Profile,
```

Comment:
Please don't put the `Profile` on the property's metadata. We can identify the property's scope elsewhere, just like we identify the project to which a project property exists. I'd like to have as much similarity between the property type used for launch profiles and that used for project properties as possible, to maximise code re-use. In the project properties we have page/category, which are fine for launch profiles too. I don't think we'd show navigation for launch profiles (though we could have category headers).

As mentioned elsewhere, the client will treat each launch profile as its own entity, sealed off from any other. Ideally the client receives the initial state and any updates for each profile individually.

Reply:
That's fine. I can switch this so a profile consists of profile metadata (the `ProfileType` you laid out elsewhere) and a set of properties.

I don't think we need the concept of a page separate from the profile--any given profile will only have one "page" of properties (possibly organized into categories, of course).